### PR TITLE
Separate map/filter/sort lambdas from normal lambdas

### PR DESCRIPTION
**Pull request by cgccuser**
My "fix" for map lambdas not being taken as single elements earlier actually broke this. Now I've made a `LambdaOp` class that's separate from the `Lambda` class. Let's hope it works properly now.

Also, mypy found over a hundred errors :(.

### CODE REVIEW

File: tests/test_complex.py [hunk 1]
```diff
@@ -72,3 +74,9 @@ def test_map_lambda_as_element():
     """Test that a map lambda is held as a single element"""
     stack = run_vyxal("⁽ƛ1+;M", inputs=[[[1, 2], [3, 4]]])
     assert stack[-1] == [[2, 3], [4, 5]]
+
+
+def test_vectorise_map_lambda():
```
Comment (chunkybanana) [inline tests/test_complex.py:79]: Maybe add tests for filter/sort as well?

File: vyxal/elements.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def process_element(
     if arity:
         arguments = ["third", "rhs", "lhs"][-arity:]
     else:
-        arguments = "_"
+        arguments = ["_"]
```
Comment (chunkybanana) [inline vyxal/elements.py:50]: What does this do?

File: vyxal/helpers.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def collect_until_false(
         val = safe_apply(function, val, ctx=ctx)
 
 
-def concat(vec1: VyList, vec2: VyList, ctx: Context = None) -> VyList:
```
Comment (chunkybanana) [inline vyxal/helpers.py:59]: Any particular reason for this?

**Review by chunkybanana — APPROVED** [review]

File: vyxal/elements.py (hunk 2 above)
Comment (cgccuser) [reply vyxal/elements.py:50]: Makes mypy happy

File: vyxal/helpers.py (hunk 3 above)
Comment (cgccuser) [reply vyxal/helpers.py:59]: Making mypy happy

File: tests/test_complex.py (hunk 1 above)
Comment (cgccuser) [reply tests/test_complex.py:79]: Good idea but i is 2 lzy

**Review by Lyxal — APPROVED** [review]
lgtm

**Pull request merged**